framework/src/base/Assembly.C

```diff
@@ -1436,35 +1491,12 @@ Assembly::reinitNeighbor(const Elem * neighbor, const std::vector<Point> & refer
     fe->attach_quadrature_rule(qrule);
     fe->reinit(neighbor);
 
-    // set the coord transformation
-    _coord_neighbor.resize(qrule->n_points());
-    Moose::CoordinateSystemType coord_type =
-        _subproblem.getCoordSystem(_current_neighbor_subdomain_id);
-    unsigned int rz_radial_coord = _subproblem.getAxisymmetricRadialCoord();
     const std::vector<Real> & JxW = fe->get_JxW();
-    const std::vector<Point> & q_points = fe->get_xyz();
+    MooseArray<Point> q_points;
+    q_points.shallowCopy(const_cast<std::vector<Point> &>(fe->get_xyz()));
 
-    switch (coord_type)
-    {
-      case Moose::COORD_XYZ:
-        for (unsigned int qp = 0; qp < qrule->n_points(); qp++)
-          _coord_neighbor[qp] = 1.;
-        break;
-
-      case Moose::COORD_RZ:
-        for (unsigned int qp = 0; qp < qrule->n_points(); qp++)
-          _coord_neighbor[qp] = 2 * M_PI * q_points[qp](rz_radial_coord);
-        break;
-
-      case Moose::COORD_RSPHERICAL:
-        for (unsigned int qp = 0; qp < qrule->n_points(); qp++)
-          _coord_neighbor[qp] = 4 * M_PI * q_points[qp](0) * q_points[qp](0);
-        break;
-
-      default:
-        mooseError("Unknown coordinate system");
-        break;
-    }
+    setCoordinateTransformation<RESIDUAL>(
```

Comment:
This only matters for <RESIDUAL>?

Reply:
The template argument here is only to help the compiler know what function you are calling because it's apparently not smart enough to infer the template argument from the function arguments. But yes you've hit on something...we've never supported an AD version of `_coord_neighbor` so if someone was using a displaced mesh and was using `_coord_neighbor`, they would be missing derivative information. But I don't actually believe `_coord_neighbor` nor `_neighbor_volume` is used anywhere in residual/Jacobian computing objects. I see it used in some user objects and postprocessors but nowhere else.

It is not used in mortar...